File: cs50/cs50x/cash/cash.c
#include <cs50.h>
#include <stdio.h>

int get_cents(void);
int calculate_quarters(int cents);
int calculate_dimes(int cents);
int calculate_nickels(int cents);
int calculate_pennies(int cents);

int main(void)
{
    // Ask how many cents the customer is owed
    int cents = get_cents();

    // Calculate the number of quarters to give the customer
    int quarters = calculate_quarters(cents);
    cents = cents - quarters * 25;

    // Calculate the number of dimes to give the customer
    int dimes = calculate_dimes(cents);
    cents = cents - dimes * 10;

    // Calculate the number of nickels to give the customer
    int nickels = calculate_nickels(cents);
    cents = cents - nickels * 5;

    // Calculate the number of pennies to give the customer
    int pennies = calculate_pennies(cents);
    cents = cents - pennies * 1;

    // Sum coins
    int coins = quarters + dimes + nickels + pennies;

    // Print total number of coins to give the customer
    printf("%i\n", coins);
}

int get_cents(void)
{
    // Asks user for inital value
    int change;
    do
    {
        change = get_int("Change owed: ");
    }
    while (change < 0);
    return change;
}

int calculate_quarters(int cents)
{
    int change = cents;
    int count = 0;

    if (cents >= 25)
    {
        do
        {
            change -= 25;
            count++;
        }
        while (change >= 25);
    }
    else
    {
        count = 0;
    }

    printf("25: %i\n", count);
    return count;
}

int calculate_dimes(int cents)
{
    int change = cents;
    int count = 0;

    if (cents >= 10)
    {
        do
        {
            change -= 10;
            count++;
        }
        while (change >= 10);
    }
    else
    {
        count = 0;
    }

    printf("10: %i\n", count);
    return count;
}

int calculate_nickels(int cents)
{
    int change = cents;
    int count = 0;

    if (cents >= 5)
    {
        do
        {
            change -= 5;
            count++;
        }
        while (change >= 5);
    }
    else
    {
        count = 0;
    }

    printf("05: %i\n", count);
    return count;
}

int calculate_pennies(int cents)
{
    int change = cents;
    int count = 0;


    if (cents >= 1)
    {
        do
        {
            change -= 1;
            count++;
        }
        while (change >= 1);
    }
    else
    {
        count = 0;
    }

    printf("01: %i\n", count);
    return count;
}

File: cs50/cs50x/sentimental-cash/cash.py
import cs50


def main():
    # Get input from user
    while True:
        dollars = cs50.get_float("Change Owed: ")
        dollars = dollars * 100
        if dollars > 0:
            break

    coins = 0

    # Get number of quarters
    quarter = quarters(dollars)
    dollars -= quarter * 25
    coins += quarter

    # Get number of dimes
    dime = dimes(dollars)
    dollars -= dime * 10
    coins += dime

    # Get number of nickels
    nickel = nickels(dollars)
    dollars -= nickel * 5
    coins += nickel

    # Get number of pennies
    penny = pennies(dollars)
    dollars -= penny * 1
    coins += penny

    print(coins)


def quarters(num):
    count = 0
    while num >= 25:
        num -= 25
        count += 1
    # print("25:",count)
    return count


def dimes(num):
    count = 0
    while num >= 10:
        num -= 10
        count += 1
    # print("10:",count)
    return count


def nickels(num):
    count = 0
    while num >= 5:
        num -= 5
        count += 1
    # print("5:",count)
    return count


def pennies(num):
    count = 0
    while num >= 1:
        num -= 1
        count += 1
    # print("1:",count)
    return count


main()
